aws/resource_aws_connect_instance.go

```diff
@@ -0,0 +1,284 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/connect"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/connect/waiter"
+)
+
+var resourceConnectInstanceAttributesMapping = map[string]string{
+	connect.InstanceAttributeTypeInboundCalls:          "inbound_calls_enabled",
+	connect.InstanceAttributeTypeOutboundCalls:         "outbound_calls_enabled",
+	connect.InstanceAttributeTypeContactflowLogs:       "contact_flow_logs_enabled",
+	connect.InstanceAttributeTypeContactLens:           "contact_lens_enabled",
+	connect.InstanceAttributeTypeAutoResolveBestVoices: "auto_resolve_best_voices",
+	connect.InstanceAttributeTypeUseCustomTtsVoices:    "use_custom_tts_voices",
+	connect.InstanceAttributeTypeEarlyMedia:            "early_media_enabled",
+}
+
+func resourceAwsConnectInstance() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAwsConnectInstanceCreate,
+		ReadContext:   resourceAwsConnectInstanceRead,
+		UpdateContext: resourceAwsConnectInstanceUpdate,
+		DeleteContext: resourceAwsConnectInstanceDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"identity_management_type": {
```

Comment:
Lets alphabetize the list

Reply:
was kinda thinking of that but I am considering keeping it in the same order for the API call definition. Happy to change it if it's a standard.